autofs-5.0.4 - fix double free in expire_proc()

From: Ian Kent <raven@themaw.net>

In state.c:expire_proc() the function expire_proc_cleanup() is called
which frees the parameter structure but automount frees this again in
the following line.
---

 CHANGELOG      |    1 +
 daemon/state.c |    1 -
 2 files changed, 1 insertions(+), 1 deletions(-)


diff --git a/CHANGELOG b/CHANGELOG
index 2cb35dc..8860b2c 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -28,6 +28,7 @@
 - add nfs mount protocol default configuration option.
 - fix bad token declaration in master map parser.
 - fix return start status on fail.
+- fix double free in expire_proc().
 
 4/11/2008 autofs-5.0.4
 -----------------------
diff --git a/daemon/state.c b/daemon/state.c
index 606743b..417fde1 100644
--- a/daemon/state.c
+++ b/daemon/state.c
@@ -298,7 +298,6 @@ static enum expire expire_proc(struct autofs_point *ap, int now)
 		error(ap->logopt,
 		      "expire thread create for %s failed", ap->path);
 		expire_proc_cleanup((void *) ea);
-		free(ea);
 		return EXP_ERROR;
 	}
 	ap->exp_thread = thid;
